src/components/ColorPicker/index.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { useEffect, useState } from "react";
+import { ActionCreator } from "redux";
+import { connect } from "react-redux";
+import { Popover, Tooltip } from "antd";
+import { HexColorInput, HexColorPicker } from "react-colorful";
+import classNames from "classnames";
+
+import selectionStateBranch from "../../state/selection";
+import { AGENT_COLORS } from "../../containers/ViewerPanel/constants";
+import {
+    ColorChangesMap,
+    SetColorChangesAction,
+    SetRecentColorsAction,
+} from "../../state/selection/types";
+
+import styles from "./style.css";
+
+interface ColorPickerProps {
+    initialColor: string;
+    agentName: string;
+    tags: string[];
+    recentColors: string[];
+    isOpen: boolean;
+    closePopover: () => void;
+    setColorChanges: ActionCreator<SetColorChangesAction>;
+    setRecentColors: ActionCreator<SetRecentColorsAction>;
+}
+
+const ColorPicker = ({
+    initialColor,
+    agentName,
+    tags,
+    recentColors,
+    isOpen,
+    closePopover,
+    setColorChanges,
+    setRecentColors,
+}: ColorPickerProps) => {
+    const [color, setColor] = useState(initialColor);
+
+    const handleColorChange = (color: string) => {
+        const colorChanges: ColorChangesMap = {
+            agents: { [agentName]: tags },
+            color: color,
+        };
+        setColorChanges(colorChanges);
+        updateRecentColors(color);
```

Comment:
I agree that it's silly to capture all those micro color changes, my instinct is to let users store colors from within a session, rather than when the popup is closed, but again I'm fairly neutral. Will definitely implement one or the other change.

Reply:
Pushed a commit that implements debounce, thanks for the suggestion, less of a party when you drag the pointer around, but much more reasonable.